### PR TITLE
misc: Update email regex to be consistent with the frontend

### PR DESCRIPTION
**Pull request by rsempe**
This PR aims to find a better approach for email validation, and make sure the FE uses the same.

### CODE REVIEW

File: app/validators/email_validator.rb [hunk 1]
```diff
@@ -1,13 +1,17 @@
 # frozen_string_literal: true
 
 class EmailValidator < ActiveModel::EachValidator
+  # Taken from HTML spec: https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address.
+  # Same frontend regexp used on the frontend: https://github.com/jquense/yup/blob/master/src/string.ts#L22.
+  EMAIL_REGEXP = /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/
```
Comment (jdenquin) [inline app/validators/email_validator.rb:6]: Style/RegexpLiteral: Use `%r` around regular expression. (https://rubystyle.guide#percent-r)

**Review by ansmonjol — APPROVED** [review]

**Review by vincent-pochet — APPROVED** [review]

File: app/validators/email_validator.rb [hunk 2]
```diff
@@ -1,13 +1,20 @@
 # frozen_string_literal: true
 
 class EmailValidator < ActiveModel::EachValidator
+  # Taken from HTML spec: https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address.
+  # Same frontend regexp used on the frontend: https://github.com/jquense/yup/blob/master/src/string.ts#L22.
+  EMAIL_REGEXP = %r{
```
Comment (ansmonjol) [inline app/validators/email_validator.rb:6]: There is an issue with the FE validation package. They decided to change this regex (you're using here) and it does not match the correct RFC 😞 

A good but ugly one we can use is the following, and it seems to map our expected usages

```
/(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/g
```
https://www.regextester.com/115911

WDYT ?

Comment (ansmonjol) [reply app/validators/email_validator.rb:6]: Here is the change PR: https://github.com/jquense/yup/commit/440db3e6177d25c06be76995a1deff6e25a90c10#diff-bbade37cef19cf470364082a01329d9fbf208a96862818d50ea97602767d7a48

We can easilly override this validation in our package.

File: app/models/customer.rb [hunk 3]
```diff
@@ -39,6 +39,7 @@ class Customer < ApplicationRecord
 
   validates :country, country_code: true, unless: -> { country.nil? }
   validates :document_locale, language_code: true, unless: -> { document_locale.nil? }
+  validates :email, email: true, if: :email?
```
[the remark this answers is not in the document]
Comment (ansmonjol) [reply app/models/customer.rb:42]: Sure ! Thanks for the hint

**Review by ansmonjol — APPROVED** [review]

**Pull request merged**